lib/rules/no-redundant-landmark-role.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Use of redundant or invalid role on <${element}> detected. If a landmark element is used, any role provided will either be redundant or incorrect.`;
+}
+
+const LANDMARK_ROLES = [
+  {
+    name: 'header',
+    role: 'banner',
+  },
+  {
+    name: 'main',
+    role: 'main',
+  },
+  {
+    name: 'aside',
+    role: 'complementary',
+  },
+  {
+    name: 'form',
+    role: 'search',
+    allow: true,
+  },
+  {
+    name: 'form',
+    role: 'form',
+  },
+  {
+    name: 'nav',
+    role: 'navigation',
+  },
+  {
+    name: 'footer',
+    role: 'contentinfo',
+    allow: true,
+  },
+];
+
+module.exports = class NoRedundantLandmarkRole extends Rule {
+  visitor() {
+    return {
+      ElementNode(node) {
+        const hasRoleAttribute = AstNodeInfo.hasAttribute(node, 'role');
+
+        if (!hasRoleAttribute) {
+          return;
+        }
+
+        const landmarkElements = LANDMARK_ROLES.map((e) => e.name);
+        if (!landmarkElements.includes(node.tag)) {
+          return;
+        }
+
+        const roleValue = AstNodeInfo.elementAttributeValue(node, 'role');
```

Comment:
We should probably short circuit if the `roleValue` here is not a static string (or `typeof roleValue !== 'string'`).